api/meetings/serializers.py

```diff
@@ -0,0 +1,213 @@
+        'self': 'get_absolute_url',
+        'html': 'get_absolute_html_url',
+        'download': 'get_download_link',
+    })
+
+    def get_author(self, obj):
+        contrib_queryset = obj.contributor_set.filter(visible=True).order_by('_order')
+        if contrib_queryset:
+            return contrib_queryset.first().user
+        return None
+
+    def get_author_id(self, obj):
+        # Author guid is annotated on queryset in ListView
+        if getattr(obj, 'author_id', None):
+            return obj.author_id
+        else:
+            author = self.get_author(obj)
+            return author._id if author else None
+
+    def get_author_name(self, obj):
+        """
+        Returns the first bibliographic contributor's family_name if it exists.
+        Otherwise, return its fullname.
+        """
+        if getattr(obj, 'author_name', None):
+            # Field is annotated on queryset in ListView for filtering purposes
+            return obj.author_name
+        else:
+            author = self.get_author(obj)
+            if author:
+                return author.family_name if author.family_name else author.fullname
+            return None
+
+    def get_meeting_category(self, obj):
+        """
+        Returns the existance of a certain tag on the node.  If the first submission type tag exists,
+        return that.  Otherwise, return the second submission type tag as a default.
+        """
+        if getattr(obj, 'meeting_category', None):
+            # Field is annotated on queryset in ListView for filtering purposes
+            return obj.meeting_category
+        else:
+            meeting = self.context['meeting']
+            submission1_name = meeting.field_names.get('submission1')
+            submission2_name = meeting.field_names.get('submission2')
+            submission_tags = obj.tags.values_list('name', flat=True)
+            return submission1_name if submission1_name in submission_tags else submission2_name
+
+    def get_download_count(self, obj):
+        """
+        Return the download counts of the first osfstorage file
+        """
+        if getattr(obj, 'download_count', None):
+            return obj.download_count or 0
+        else:
+            node_ct = ContentType.objects.get_for_model(AbstractNode).id
+            with connection.cursor() as cursor:
+                cursor.execute(
+                    """
+                    SELECT P.total
+                    FROM osf_basefilenode F, osf_pagecounter P
+                    WHERE (F.type = 'osf.osfstoragefile'
+                         AND F.provider = 'osfstorage'
+                         AND F.target_content_type_id = %s
+                         AND F.target_object_id = %s
+                         AND P._id = 'download:' || %s || ':' || F._id)
+                    ORDER BY F.id ASC
+                    LIMIT 1;
+                """, [node_ct, obj.id, obj._id],
+                )
+                result = cursor.fetchone()
```

Comment:
It returns the next row in the result set in tuple form.  Here my SQL query is only returning the one count - but say I had 2 downloads.  Tuple `(2,)` would be returned.  Versus cursor.fetchall(), which returns all rows in the result set, as a list of tuples, would return - `[(2,)]`.

Reply:
As far as keeping this DRY, the SQL is specific to getting the download counts off the first osfstorage file on the project.  There's something similar as part of a much larger query on the corresponding v1 views, but I don't see a reason to consolidate with code that's going to be deleted shortly.  I thought about making this a property on the Node model, but it's really specific to meetings and seemed like it didn't go there.

Reply:
For the DRYing, that makes total sense. But my question on the `fetchone()` is why use cursor rather than just running the query? Is it _just_ to get the data as a tuple, and/or is this cursor something different from using a cursor for pagination, or something else?

Reply:
I'm doing this:
https://docs.djangoproject.com/en/2.2/topics/db/sql/#executing-custom-sql-directly

Reply:
Awesome, thank you. I’d seen `raw()` but I hadn’t looked closely at the other ways, since I’ve been just doing QuerySet stuff.